src/Symfony/Component/Validator/Tests/Constraints/EmailValidatorTest.php

```diff
@@ -86,4 +88,28 @@ public function getInvalidEmails()
             array('example@localhost'),
         );
     }
+
+    /**
+     * @dataProvider getHostnamesAndEmailAddresses
+     */
+    public function testHostnameIsProperlyParsed($hostnameAndEmail)
+    {
+        $method = new ReflectionMethod($this->validator, 'extractHostname');
+        $method->setAccessible(true);
```

Comment:
I can't see any realistic way of testing this code if it remains in the `validate` method, because the test would then unnecessarily depend on network connectivity and the `checkdnsrr` function.